mod/cornercutter/ModFeature/ModSetting/ItemBlocker.cs

```diff
@@ -0,0 +1,70 @@
+﻿using cornercutter.Enum;
+using cornercutter.ModLoader;
+using HarmonyLib;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace cornercutter.ModFeature.ModSetting
+{
+    [HarmonyPatch(typeof(Player), nameof(Player.GrabItem))]
+    class ItemBlocker
+    {
+        static readonly List<CallerInfo> SkillsGrantingItems = new List<CallerInfo>()
+        {
+            new CallerInfo { MethodName = "MoveNext", ClassName = "<StartWithBento>d__8" },
+            new CallerInfo { MethodName = "MoveNext", ClassName = "<SpawnBomb>d__5" }
+        };

```

Comment:
Out of curiosity, why are these class names kind of weird?

Reply:
Good question! This is because these are references to compiler generated types (don't worry, these are deterministic!) - in this case, the method adding the item is a Unity coroutine using IEnumerator. An example is below. 
![image](https://user-images.githubusercontent.com/9399879/203306662-567216f4-88fc-45fb-800e-46bc19e7972d.png)

For more info, see https://stackoverflow.com/a/32698545

Reply:
Ahhh, that's very interesting